src/Rule/Url.php

```diff
@@ -77,11 +77,7 @@ public function getName(): string
 
     public function getPattern(): string
     {
-        if (str_contains($this->pattern, '{schemes}')) {
-            return str_replace('{schemes}', '((?i)' . implode('|', $this->validSchemes) . ')', $this->pattern);
-        }
-
-        return $this->pattern;
+        return strtr($this->pattern, ['{schemes}' => '((?i)' . implode('|', $this->validSchemes) . ')']);
```

Comment:
Why `strtr` instead of `str_replace` ?

Reply:
For 1 replacement it might be a better choice. Brought back `str_replace`, but I think `str_contains` it's unnecessary anyway.